### PR TITLE
[WIP] Add color legend for the broadcasts relationship graph

### PR DESCRIPTION
**Pull request by tbaraza**

### CODE REVIEW

File: frontend/app/components/visualize/graph/graph-visualization.js [hunk 1]
```diff
@@ -12,8 +12,15 @@ export default Component.extend({
...
-    let color = d3.scaleOrdinal(d3.schemeCategory10);
+    let color = d3.scaleOrdinal()
+      .domain([0, 1, 2, 3, 4])
+      .range([
+        d3.rgb("#016936"),
+        d3.rgb("#FE9A76"),
+        d3.rgb("#0E6EB8"),
+        d3.rgb("#B03060"),
+        d3.rgb("#B413EC")
+      ]);
```
Comment (tbaraza) [inline frontend/app/components/visualize/graph/graph-visualization.js]: Hey, @roschaefer I have set up custom colors. Your feedback will be highly appreciated. Also, I need the mapping of the different groups to the type of the broadcasts e.g broadcast name for  group 0 etc

File: frontend/config/environment.js [hunk 2]
```diff
@@ -17,9 +17,6 @@ module.exports = function(environment) {
         domain: (process.env.AUTH0_DOMAIN || "rundfunk-testing.eu.auth0.com")
       },
     },
-    'ember-d3': {
-      bundle: true
-    },
```
Comment (roschaefer) [inline frontend/config/environment.js]: @tbaraza sorry for removing the colour legend based on HTML. I found out that if we remove this line here, we can import 3rd party plugins. There is already a colour legend plugin for d3 out there. It renders the legend into the svg. So, in this branch #759 I did a refactoring and removed the global import of `d3`. Then on your branch I added `d3-svg-legend` as a dependency and inserted the legend into `graph-visualization` component.

File: frontend/app/routes/visualize/graph.js [hunk 3]
```diff
@@ -21,9 +22,13 @@ export default Route.extend(ResetScrollPositionMixin, {
...
       }
 
-      return fetch(url, {headers: headers}).then(function(response) {
-        return response.json();
-      });
+      const model = {
+        graph: fetch(url, {headers: headers}).then(function(response) {
+          return response.json();
+        }),
+        media: this.store.findAll('medium')
```
Comment (roschaefer) [inline frontend/app/routes/visualize/graph.js]: That's where the data for the labels comes from: We query all records of type `medium`. It is passed to the `graph-visualization` component and in there we use the `name` attribute as a label.

**Comment by roschaefer** [comment]
@tbaraza I haven't found out how to reload the page when the user changes the language. Could you maybe research it? Because the translated names for the `medium` types come from the backend, we have to reload the `model` when the user changes the language.

Here's the relevant plugin: https://github.com/ember-intl/ember-intl
And maybe there is some hint in the ember guides how to reload the `model` hook? https://guides.emberjs.com/release/

**Comment by roschaefer** [comment]
@tbaraza we could refactor this component here: https://github.com/roschaefer/rundfunk-mitbestimmen/blob/master/frontend/app/components/visualize/bubble-chart-legend.js

It is a lot of code which we can now entirely replace with `colorLegend`. Does that sound like a nice refactoring? Learning how to use `d3` guaranteed!

**Pull request merged**